rust/integration-tests/src/lib.rs

```diff
@@ -0,0 +1,121 @@
+use anyhow::Context;
+use aptos_protos::transaction::v1::Transaction;
+use diesel::{pg::PgConnection, sql_query, Connection, RunQueryDsl};
+use itertools::Itertools;
+use processor::{
+    processors::{ProcessorConfig, ProcessorTrait},
+    utils::database::{new_db_pool, run_pending_migrations},
+    worker::build_processor_for_testing,
+};
+use testcontainers::{
+    core::{IntoContainerPort, WaitFor},
+    runners::AsyncRunner,
+    ContainerAsync, GenericImage, ImageExt,
+};
+
+mod event_processor;
+
+/// The test context struct holds the test name and the transaction batches.
+pub struct TestContext {
+    pub transaction_batches: Vec<Transaction>,
```

Comment:
Instead of just a Vec<Transaction>, we could have it return Map<TransactionVersion, Transaction>. That way, the test can assert between the json txn and the DB rows on specific txn data.

Reply:
not quite following why we have to hardcode values in the test case.  could you elaborate more?

Reply:
What I was suggesting is to provide the input test transactions to each test case so that you could have assertions like:
```
assert(txn.events.0.event_address, db.event_address)
```
But actually I think it might be better to hardcode the asserted values in the test case, because the proto schema and output could change

Reply:
Thanks for elaborating it! I see the value in hardcoding the asserted values to ensure stability, especially considering that the proto schema and output might change over time. However, I also think that providing some fields directly from the input transactions can be beneficial, as it allows us to verify that the processing logic correctly handles data anyways.

so to balance this out, i propose this:

1. For fields that are unlikely to change frequently or are critical, I'll hardcode the expected values in the test cases like indexed_type, address, creation_number.

2. For fields that are easily derived from the input transactions, I'll pull the expected values directly from the input transactions. This way, we can still validate that the processing logic correctly parses and stores these fields.

how does that sound to you?

Reply:
Sounds good! I like giving the test cases the option of whether to hardcode vs derive from input txn's. Deriving from input txn's also makes the test cases more versatile. For example, the same test can be used to test transfers A -> B txn and also C -> D because the test case can derive the from/to address instead of hardcoding the transfer addresses in the test.